openquake/hazardlib/shakemap/conditioned_gmfs.py

```diff
@@ -0,0 +1,988 @@
+        return result, sig, eps
+
+    def _compute(self, mu_Y, cov_WY_WY, cov_BY_BY, imt, num_events, rng):
+        if self.cmaker.truncation_level == 0:
+            gmf = exp(mu_Y, imt)
+            gmf = gmf.repeat(num_events, axis=1)
+            inter_sig = 0
+            inter_eps = [numpy.zeros(num_events)]
+        else:
+            cov_Y_Y = cov_WY_WY + cov_BY_BY
+            # cov_Y_Y_near = cov_nearest(
+            #     cov_Y_Y, method="clipped", threshold=1e-15, n_fact=100, return_all=False
+            # )
+            # cov_Y_Y_near = numpy.nan_to_num(cov_Y_Y_near).clip(min=0)
+            gmf = exp(
+                rng.multivariate_normal(
+                    mu_Y.flatten(),
+                    cov_Y_Y,
+                    size=num_events,
+                    check_valid="warn",
+                    tol=1e-5,
+                    method="eigh",
+                ),
+                imt,
+            ).T
+            inter_sig = 0
+            inter_eps = 0
+        return gmf, inter_sig, inter_eps  # shapes (N, E), 1, E
+
+
+def get_conditioned_mean_and_covariance(
+    rupture,
+    gmm,
+    station_sitecol,
+    station_data,
+    observed_imt_strs,
+    target_sitecol,
+    target_imts,
+    spatial_correl,
+    cross_correl_between,
+    cross_correl_within,
+    maximum_distance,
+):
+    gmm_name = gmm.__class__.__name__
+    if gmm_name == "ModifiableGMPE":
+        gmm_name = gmm.gmpe.__class__.__name__
+
+    if gmm.DEFINED_FOR_STANDARD_DEVIATION_TYPES == {StdDev.TOTAL}:
+        modifications = list(gmm.kwargs.keys())
+        if not (
+            type(gmm).__name__ == "ModifiableGMPE"
+            and "add_between_within_stds" in modifications
+        ):
+            raise NoInterIntraStdDevs(gmm)
+
+    observed_imtls = {
+        imt_str: [0] for imt_str in observed_imt_strs if imt_str not in ["MMI", "PGV"]
+    }
+    observed_imts = sorted([imt.from_string(imt_str) for imt_str in observed_imtls])
+    cmaker_D = ContextMaker(
+        rupture.tectonic_region_type,
+        [gmm],
+        dict(
+            truncation_level=0,
+            imtls=observed_imtls,
+            maximum_distance=maximum_distance,
+        ),
+    )
+
+    gc_D = GmfComputer(rupture, station_sitecol, cmaker_D)
+    mean_stds = cmaker_D.get_mean_stds([gc_D.ctx])[:, 0]
+    station_locs_filtered = (
+        numpy.argwhere(numpy.isin(station_sitecol.sids, gc_D.ctx.sids)).ravel().tolist()
+    )
+    station_sitecol_filtered = station_sitecol.filtered(station_locs_filtered)
+    station_data_filtered = station_data.iloc[station_locs_filtered]
+    num_station_sites = len(station_sitecol_filtered)
+
+    # (4, G, M, N): mean, StdDev.TOTAL, StdDev.INTER_EVENT, StdDev.INTRA_EVENT; G gsims, M IMTs, N sites/distances
+    for i, imt_i in enumerate(observed_imts):
+        station_data_filtered.loc[:, imt_i.string + "_" + "median"] = mean_stds[0, i, :]
```

Comment:
Are you sure that it is "median" and not "mean"? A few lines below I see a line
```python
station_data_filtered[[c_imt.string + "_mean" for c_imt in conditioning_imts]]
```

Reply:
Yes. The "median" values here are the ones predicted by the GMMs. The "mean" values later on are the ones observed at the recording stations.